app/src/main/java/pl/droidevs/books/addBook/AddBookActivity.java

```diff
@@ -0,0 +1,144 @@
+package pl.droidevs.books.addBook;
+
+import android.arch.lifecycle.ViewModelProvider;
+import android.arch.lifecycle.ViewModelProviders;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.constraint.ConstraintLayout;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.Spinner;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.request.RequestOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import butterknife.OnTextChanged;
+import dagger.android.AndroidInjection;
+import pl.droidevs.books.R;
+import pl.droidevs.books.model.Book;
+
+import static com.bumptech.glide.Priority.HIGH;
+
+public class AddBookActivity extends AppCompatActivity {
+
+    @BindView(R.id.addBookConstraintLayout)
+    ConstraintLayout container;
+
+    @BindView(R.id.coverImageUrlEditText)
+    EditText coverUrlEditText;
+
+    @BindView(R.id.coverImageView)
+    ImageView coverImageView;
+
+    @BindView(R.id.categorySpinner)
+    Spinner categorySpinner;
+
+    @BindView(R.id.titleEditText)
+    EditText titleEditText;
+
+    @BindView(R.id.authorEditText)
+    EditText authorEditText;
+
+    @BindView(R.id.descriptionEditText)
+    EditText descriptionEditText;
+
+    @Inject
+    ViewModelProvider.Factory viewModelFactory;
+
+    AddBookViewModel addBookViewModel;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_book);
+
+        AndroidInjection.inject(this);
+        ButterKnife.bind(this);
+
+        setupViewModel();
+        setupSpinner();
+    }
+
+    private void setupViewModel() {
+        this.addBookViewModel = ViewModelProviders.of(this, viewModelFactory).get(AddBookViewModel.class);
+        this.addBookViewModel.wasAddingSuccessful()
+                .observe(this, wasAddingSuccessful -> {
+                    if (wasAddingSuccessful) {
+                        finish();
+                    } else {
+                        displaySnackBar(R.string.saving_book_error);
+                    }
+        });
+    }
+
+    private void displaySnackBar(int messageResource) {
+        Snackbar.make(container, getString(messageResource), Snackbar.LENGTH_SHORT).show();
```

Comment:
fajnie by to było kiedyś przenieść do jakiejś ogólnej klasy która obsługuje Snackbar, bo taka metoda pojawi się pewnie w każdym Activity
poza tym w metodzie make() możesz podać ID resource tekstu bezpośrednio, nie trzeba wrzucać Sstringa: Snackbar.make(container, messageResource, Snackbar.LENGTH_SHORT).show();